compiler/crates/relay-compiler/src/config.rs

```diff
@@ -408,7 +409,7 @@ Example file:
             header: config_file.header,
             codegen_command: config_file.codegen_command,
             load_saved_state_file: None,
-            generate_extra_artifacts: None,
+            generate_extra_artifacts: Some(Box::new(default_generate_extra_artifacts_fn)),
```

Comment:
My concern is there could be existing commands at Meta that shouldn't generate extra artifacts. Could this be configured just for the OSS binary?

Reply:
I'm not sure how I could configure this just for the OSS binary. Are there any existing examples for such a case?

Reply:
At Atlassian we've customised our relay-bin to [output persisted mocks](https://github.com/facebook/relay/issues/4465). 


Easy to tweak so that we chain these calls since we're already customising that - but since this is a new feature, potentially it can be a different config (or, feature flagged rather than a config) and done with an extra `artifacts.extend` just before the `generate_extra_artifacts_fn` runs [here](https://github.com/facebook/relay/blob/9f135b8f92a750c7bc79c08bda0c149539ac0bc8/compiler/crates/relay-compiler/src/build_project/mod.rs#L360-L370)?

Reply:
Good point @tyao1 

@tobias-tengler I think you can move that config to `relay-bin`:

```
diff --git a/compiler/crates/relay-bin/src/main.rs b/compiler/crates/relay-bin/src/main.rs
index f832244070dd..51bab9e93822 100644
--- a/compiler/crates/relay-bin/src/main.rs
+++ b/compiler/crates/relay-bin/src/main.rs
@@ -18,6 +18,7 @@ use intern::string_key::Intern;
 use log::error;
 use log::info;
 use relay_compiler::build_project::artifact_writer::ArtifactValidationWriter;
+use relay_compiler::build_project::generate_extra_artifacts::default_generate_extra_artifacts_fn;
 use relay_compiler::compiler::Compiler;
 use relay_compiler::config::Config;
 use relay_compiler::errors::Error as CompilerError;
@@ -296,6 +297,8 @@ async fn handle_compiler_command(command: CompileCommand) -> Result<(), Error> {
         );
     }
 
+    config.generate_extra_artifacts = Some(Box::new(default_generate_extra_artifacts_fn));
+
     let compiler = Compiler::new(Arc::new(config), Arc::new(ConsoleLogger));
 
     if command.watch {
diff --git a/compiler/crates/relay-compiler/src/config.rs b/compiler/crates/relay-compiler/src/config.rs
index b42255bb86a4..f5861eeaf6a7 100644
--- a/compiler/crates/relay-compiler/src/config.rs
+++ b/compiler/crates/relay-compiler/src/config.rs
@@ -57,7 +57,6 @@ use watchman_client::pdu::ScmAwareClockData;
 
 use crate::build_project::artifact_writer::ArtifactFileWriter;
 use crate::build_project::artifact_writer::ArtifactWriter;
-use crate::build_project::generate_extra_artifacts::default_generate_extra_artifacts_fn;
 use crate::build_project::generate_extra_artifacts::GenerateExtraArtifactsFn;
 use crate::build_project::get_artifacts_file_hash_map::GetArtifactsFileHashMapFn;
 use crate::build_project::AdditionalValidations;
@@ -409,7 +408,7 @@ Example file:
             header: config_file.header,
             codegen_command: config_file.codegen_command,
             load_saved_state_file: None,
-            generate_extra_artifacts: Some(Box::new(default_generate_extra_artifacts_fn)),
+            generate_extra_artifacts: None,
             generate_virtual_id_file_name: None,
             get_artifacts_file_hash_map: None,
             saved_state_config: config_file.saved_state_config,
@@ -576,7 +575,11 @@ impl fmt::Debug for Config {
         } = self;
 
         fn option_fn_to_string<T>(option: &Option<T>) -> &'static str {
-            if option.is_some() { "Some(Fn)" } else { "None" }
+            if option.is_some() {
+                "Some(Fn)"
+            } else {
+                "None"
+            }
         }
 
         f.debug_struct("Config")
```

Reply:
Moved it to relay-bin :)